astra/src/main/java/com/slack/astra/bulkIngestApi/opensearch/BulkApiRequestParser.java

```diff
@@ -49,11 +50,11 @@ public static Map<String, List<Trace.Span>> parseRequest(
    * in millis 2. Check if a field called `@timestamp` exists and parse that as a date (since
    * logstash sets that) 3. Use the current time from the ingestMetadata
    */
-  public static long getTimestampFromIngestDocument(IngestDocument ingestDocument) {
+  public static long getTimestampFromIngestDocument(Map<String, Object> sourceAndMetadata) {
 
     try {
-      if (ingestDocument.hasField("@timestamp")) {
```

Comment:
hasField basically gets sourceAndMetadata so the change shouldn't have any effect.